### ISSUE TITLE
Shader error after parallax changes

### ISSUE THREAD

**Issue by kaeza** [minetest/minetest#2866]
Putting this here so it doesn't get lost.

Commit 43fcfbfe05578d7471d40c8c087fd04e24b264b5 introduced a regression in the rendering (notice the red surfaces at the bottom of the picture):

![](http://i.imgur.com/VM4axoy.png)

This only happens at certain angles; rotating the camera makes the nodes alternate between red tint and their regular colors.

Some system info:

```
Irrlicht log: Irrlicht Engine version 1.8.1
Irrlicht log: Microsoft Windows 7 Ultimate Edition  (Build 7600)
Irrlicht log: WGL_extensions: WGL_ARB_buffer_region WGL_ARB_extensions_string WGL_ARB_make_current_read WGL_ARB_pixel_format WGL_ARB_pbuffer WGL_EXT_extensions_string WGL_EXT_swap_control WGL_ARB_pixel_format_float
Irrlicht log: Using renderer: OpenGL 2.1.0
Irrlicht log: Intel(R) G41 Express Chipset: Intel
Irrlicht log: OpenGL driver version is 1.2 or better.
Irrlicht log: GLSL version: 1.2
```

**Comment by nerzhul** [comment]
looks like a shader bug for your card

**Comment by est31** [comment]
@RealBadAngel has said its because its opengl version 2.1. Ideally, we have an additional "simple shader" that works with opengl 2.1 (still common), and perhaps even opengl ES. Then we can switch to simple shaders if we detect an older shader version.

**Comment by RealBadAngel** [comment]
please remove "bug" label. its just too OLD hardware

**Comment by kaeza** [comment]
>OLD hardware

WTF? This was clearly a regression in rendering: it did not happen before and suddenly it started to happen. It IS a bug.

**Comment by est31** [comment]
I think it is, too. We do want to support opengl 2.1.

**Comment by HybridDog** [comment]
as far as l know graphics chips/cards which support opengl 1.2 don't have shaders yet

**Comment by est31** [comment]
Ah I see, I've mistaken "Irrlicht log: OpenGL driver version is 1.2 or better." with the actual version here.

**Comment by HybridDog** [comment]
since some version, maybe 0.4.9, minetest doesn't longer work with opengl 1.3, l think,
and a new l/x/ubuntu doesn't support that graphics chips and selects the laggy mesa software rendering

**Comment by doyousketch2** [comment]
![flzganhibycqkfg_jeg5enmykxxij03xjxsq9fm65m0](https://cloud.githubusercontent.com/assets/3681804/9421559/99111302-4843-11e5-9c98-3e04dd13bda5.png)


I have 2 friends that this is happening to on the latest ver of Minetest.  They are using windows 7 laptops, Open GL 2.1, and have tried both the 32bit and 64bit versions, with or without shaders.  It looks as if things are being highlighted that aren't being pointed to.

**Comment by SivaMachina** [comment]
<blockquote> please remove "bug" label. its just too OLD hardware. <blockquote>
@RealBadAngel blaming it on old hardware is horrible excuse for a mistake or bug in coding or even saying it is not a bug at all because of "old" hardware.

**Comment by Ekdohibs** [comment]
@kaeza Do you still have this problem, since there have been quite a lot of graphics changes lately?

**Comment by kaeza** [comment]
@Ekdohibs Looks like it's partially fixed. It doesn't show up until you start enabling lots of shaders at once. Just enabling shaders and waving stuff works OK.

There's still a huge penalty in terms of FPS (15 FPS with just "shaders" looking directly down at the grass; I used to get at least 40 FPS in regular play in 0.4.11 or so.

**Comment by rubenwardy** [comment]
What OpenGL version did you use?

**Comment by paramat** [comment]
Still occuring? Closing in 1 month.

**Comment by paramat** [comment]
@kaeza

**Comment by Fixer-007** [comment]
I had the same problem on old laptop, I guess I can check too.

**Comment by devshgraphicsprogramming** [comment]
> Looks like this is due to old OpenGL version

The PArallax Shader is actually implemented as an ancient OpenGL 1.4 extensions NVidia Assembly Program instead of GLSL.
AFAIK in all versions of irrlicht, even SVN (obviously not in the OpenGL core profile branch).

**Comment by numberZero** [comment]
Obsolete. Parallax was removed entirely.
